Sources/TuistGenerator/Linter/GraphLinter.swift

```diff
@@ -223,28 +223,26 @@ public class GraphLinter: GraphLinting {
     private func lintMismatchingConfigurations(graphTraverser: GraphTraversing) -> [LintingIssue] {
         let rootProjects = graphTraverser.rootProjects()
 
-        let knownConfigurations = rootProjects.reduce(into: Set()) {
-            $0.formUnion(Set($1.settings.configurations.keys))
-        }
-
-        let projectBuildConfigurations = graphTraverser.projects.compactMap { project in
-            (name: project.value.name, buildConfigurations: Set(project.value.settings.configurations.keys))
-        }
-
-        let mismatchingBuildConfigurations = projectBuildConfigurations.filter {
-            !knownConfigurations.isSubset(of: $0.buildConfigurations)
+        var reasons = Set<String>()
+        for project in rootProjects {
+            let rootProjectConfigurations = Set(project.settings.configurations.keys)
+            let sortedRootProjectConfigurations = rootProjectConfigurations.sorted()
+
+            for target in project.targets.values {
+                for dependency in graphTraverser.allTargetDependencies(path: project.path, name: target.name) {
```

Comment:
This is not ideal as we can visit a lot of nodes multiple times.

I'd recommend using the [filterDependencies](https://github.com/tuist/tuist/blob/main/Sources/TuistCore/Graph/GraphTraverser.swift#L819) instead.

Reply:
Done – as `filterDependencies` is internal and probably shouldn't be exposed publicly, I added a new public method that returns all target dependencies of a project and uses `filterDependencies` underneath. The linter is now using this new method.

Reply:
Hopefully it's not the same as the new `func allTargetDependencies(traversingFromTargets: [GraphTarget]) -> Set<GraphTarget>` that you added while I was preparing this change. Let me know if should use that one instead

Reply:
I believe it's the same. Would you mind using that one instead?